Use ninja as a consistent build tool across OS/arch

dicta opened the pull request:
<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

Intended to address the CI failing builds in #18 and #19.

dicta committed:
Use ninja as a consistent build tool across OS/arch

ryanvolz commented:
I think once upon a time I had issues with Ninja on Windows for `volk`, but we'll see what happens. We might want to fold this into one of the arch migrations, just so we don't have an excess of builds. That depends on how much time I have in the immediate future to debug those, though, since I've got to accomplish some other things over the next few hours.

ryanvolz commented:
Oh, and I think the main issue with those arch migrations is incompatibilities with the internal `cpu_features`, but I haven't given them a detailed look. So tweaking the build tooling like this just might be the trick.

dicta committed:
Also update bld.bat

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ryanvolz committed:
Un-bump the build number

Switching to ninja doesn't effect the build output, and we'll incorporate these changes into a new build soon with the arch migrations anyway.
